File: 3111/main.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	fmt.Println(minRectanglesToCoverPoints([][]int{
		{2,1},
		{1, 0},
		{1, 4},
		{1, 8},
		{3, 5},
		{4, 6},
	}, 1))
}

func minRectanglesToCoverPoints(points [][]int, w int) int {
	sort.Slice(points, func(i, j int) bool {
		return points[i][0] < points[j][0]
	})
	bound := -1
	ans := 0 
	for _, point := range points {
		if point[0] > bound {
			bound = point[0] + w
			ans++
		}
	}
	return ans
}

File: 3111/main.rs
struct Solution{}

impl Solution {
    pub fn min_rectangles_to_cover_points(mut points: Vec<Vec<i32>>, w: i32) -> i32 {
        let mut bound = -1;
        let mut ans = 0;
        points.sort();
        for point in points {
            if point[0] > bound {
                bound = point[0] + w;
                ans += 1;
            }
        }
        ans
    }
}

fn main() {
    println!("{}", Solution::min_rectangles_to_cover_points(vec![
        vec![2,3],
        vec![1, 2],
    ], 0))
}